Plugins/GeoModel/src/GeoModelToDetectorVolume.cpp

```diff
@@ -96,93 +57,96 @@ std::shared_ptr<Experimental::DetectorVolume> convertVolume(
     // shape
     const GeoShapeSubtraction* subtractionShape =
         dynamic_cast<const GeoShapeSubtraction*>(&shape);
     const GeoShape* shapeA = subtractionShape->getOpA();
-    return convertVolume(context, *shapeA, name, transform, sensitives);
+    return package(trf, *shapeA);
   } else if (shape.typeID() == GeoPcon::getClassTypeID()) {
     // Will change in future, get bounding box for now
     double xmin{0}, xmax{0}, ymin{0}, ymax{0}, zmin{0}, zmax{0};
     const GeoPcon* pcon = dynamic_cast<const GeoPcon*>(&shape);
     pcon->extent(xmin, ymin, zmin, xmax, ymax, zmax);
-    std::shared_ptr<CuboidVolumeBounds> bounds =
-        std::make_shared<CuboidVolumeBounds>(
-            (xmax - xmin) / 2, (ymax - ymin) / 2, (zmax - zmin) / 2);
-    return Experimental::DetectorVolumeFactory::construct(
-        portalGenerator, context, name, transform, bounds,
-        Experimental::tryAllPortalsAndSurfaces());
-  }
-  if (shape.typeID() == GeoShapeShift::getClassTypeID()) {
+    bounds = std::make_shared<CuboidVolumeBounds>(
+        (xmax - xmin) / 2, (ymax - ymin) / 2, (zmax - zmin) / 2);
+  } else if (shape.typeID() == GeoShapeShift::getClassTypeID()) {
     const GeoShapeShift* shiftShape =
         dynamic_cast<const GeoShapeShift*>(&shape);
     const GeoShape* shapeOp = shiftShape->getOp();
-    GeoTrf::Transform3D newTransform = transform * shiftShape->getX();
-    return convertVolume(context, *shapeOp, name, newTransform, sensitives);
+    newTrf = trf * shiftShape->getX();
+    return package(trf, *shapeOp);
+  } else {
+    throw std::runtime_error("FATAL: Unsupported GeoModel shape");
   }
-  throw std::runtime_error("FATAL: Unsupported GeoModel shape");
+
+  Volume vol = Volume(newTrf, bounds);
+  return vol;
+}
+
+std::shared_ptr<Experimental::DetectorVolume> convertVolume(
+    const GeometryContext& context, const GeoShape& shape,
+    const std::string& name, const GeoTrf::Transform3D& transform,
+    const std::vector<GeoModelSensitiveSurface>& sensitives) {
+  // dummy volume for conversion with surfaces
+  std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>> a;
+
+  // type conversion from GeoModelSensitiveSurface to Surface
+  std::vector<std::shared_ptr<Surface>> sensSurfaces(sensitives.size());
+  std::transform(sensitives.begin(), sensitives.end(), sensSurfaces.begin(),
+                 [](const std::tuple<std::shared_ptr<GeoModelDetectorElement>,
+                                     std::shared_ptr<Surface>>& t) {
+                   return std::get<1>(t);
+                 });
+  auto portalGenerator = Experimental::defaultPortalAndSubPortalGenerator();
+  Volume vol = package(transform, shape);
+  return Experimental::DetectorVolumeFactory::construct(
+      portalGenerator, context, name, vol.transform(),
+      std::const_pointer_cast<VolumeBounds>(vol.volumeBoundsPtr()),
+      sensSurfaces, a, Experimental::tryNoVolumes(),
```

Comment:
following my previous comment, not define a at all and just put here a dummy vector since you don't want to pass volumes in your detector volume